sdk/appconfiguration/app-configuration/src/policies/throttlingRetryPolicy.ts

```diff
@@ -35,27 +41,51 @@ const StandardAbortMessage = "The operation was aborted.";
  * @internal
  */
 export class ThrottlingRetryPolicy extends BaseRequestPolicy {
-  constructor(nextPolicy: RequestPolicy, options: RequestPolicyOptions) {
+  private numberOfRetries = 0;
+  constructor(
+    nextPolicy: RequestPolicy,
+    options: RequestPolicyOptions,
+    private retryOptions: Pick<RetryOptions, "maxRetries" | "maxRetryDelayInMs"> = {}
+  ) {
     super(nextPolicy, options);
   }
 
   public async sendRequest(httpRequest: WebResource): Promise<HttpOperationResponse> {
     return this._nextPolicy.sendRequest(httpRequest.clone()).catch(async (err) => {
       if (isRestErrorWithHeaders(err)) {
-        const delayInMs = getDelayInMs(err.response.headers);
+        let delayInMs = getDelayInMs(err.response.headers);
 
         if (delayInMs == null) {
           throw err;
         }
 
+        if (
+          this.retryOptions.maxRetryDelayInMs &&
```

Comment:
can we set this to the default in the constructor default option value? Otherwise the delay can go unbounded.
```ts
    private retryOptions: Pick<RetryOptions, "maxRetries" | "maxRetryDelayInMs"> = {
      maxRetryDelayInMs: 90*1000 // is this constant defined somewhere?
      maxRetries: DEFAULT_CLIENT_RETRY_COUNT
    }
```

Reply:
This is fine for maxRetries.
I don't want to define maxRetryDelayInMs and different services may have different values.

For example, with the observations from app-config, I have seen values anywhere between 20ms and 300000ms.

Reply:
For the throttlign policy case, if the error from the service is with the delay, then we use it for delay and throw directly otherwise. I don't think there is a case of unboundedness.

maxRetryDelayInMs should only be provided by the user according to me.

Reply:
Sounds fair.  Please update the ref doc for maxRetryDelayInMs to remove the mentioning of default value.

Reply:
Updated